api/integrations/sentry/middleware.py

```diff
@@ -0,0 +1,32 @@
+import logging
+
+import sentry_sdk
+from django.conf import settings
+
+logger = logging.getLogger(__name__)
+
+
+class ForceSentryTraceMiddleware:
+    """
+    Middleware class to allow us to force Sentry traces on given requests.
+
+    To use the middleware the application must be running in an environment with the
+    FORCE_SENTRY_TRACE_KEY environment variable set. Assuming that is done,
+    simply add the `X-Force-Sentry-Trace-Key` header to any request and make sure it
+    is set to the same value as the FORCE_SENTRY_TRACE_AUTH_KEY environment variable.
+    """
+
+    FORCE_SENTRY_TRACE_HEADER = "X-Force-Sentry-Trace-Key"
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.auth_key = settings.FORCE_SENTRY_TRACE_KEY
+
+    def __call__(self, request):
+        auth = request.headers.get(self.FORCE_SENTRY_TRACE_HEADER)
+        if auth == self.auth_key:
```

Comment:
would it not be better/more efficient to check if `auth` is not None at this point?

Reply:
You mean `if auth and auth == self.auth_key` ? 

I'm not sure what we benefit from that?

Reply:
Note that I changed the settings file to: 

```python
# Sentry tracking
SENTRY_SDK_DSN = env("SENTRY_SDK_DSN", default=None)
SENTRY_TRACE_SAMPLE_RATE = env.float("SENTRY_TRACE_SAMPLE_RATE", default=1.0)
FORCE_SENTRY_TRACE_KEY = env("FORCE_SENTRY_TRACE_KEY", default=None)
if FORCE_SENTRY_TRACE_KEY:
    MIDDLEWARE.append("integrations.sentry.middleware.ForceSentryTraceMiddleware")
```

So it will never be initialised without a key.